src/server/pfs/server/master.go

```diff
@@ -87,78 +93,134 @@ func (d *driver) master(ctx context.Context) {
 	})
 }
 
-func (d *driver) manageRepos(ctx context.Context) error {
+func (d *driver) watchRepos(ctx context.Context) error {
 	ctx, cancel := pctx.WithCancel(ctx)
 	defer cancel()
-	repos := make(map[string]context.CancelFunc)
+	repos := make(map[pfsdb.RepoID]context.CancelFunc)
 	defer func() {
 		for _, cancel := range repos {
 			cancel()
 		}
 	}()
 	return consistenthashing.WithRing(ctx, d.etcdClient, path.Join(d.prefix, masterLockPath, "ring"),
 		func(ctx context.Context, ring *consistenthashing.Ring) error {
-			err := d.repos.ReadOnly(ctx).WatchF(func(ev *watch.Event) error {
-				if ev.Type == watch.EventError {
-					return ev.Err
-				}
-				key := string(ev.Key)
-				lockPrefix := path.Join("repos", key)
-				if ev.Type == watch.EventDelete {
-					if cancel, ok := repos[key]; ok {
-						if err := ring.Unlock(lockPrefix); err != nil {
-							return err
+			// watch for new repo events.
+			watcher, err := postgres.NewWatcher(d.env.DB, d.env.Listener, masterLockPath, v2_8_0.ReposChannelName)
+			if err != nil {
+				return errors.Wrap(err, "new watcher")
+			}
+			defer watcher.Close()
+			var eg errgroup.Group
+			eg.Go(func() error {
```

Comment:
I don't think we need this to be in a call to `eg.Go`.  This is the main flow, this "for select" loop should be what the master does until it errors.

Reply:
Move to the bottom, outside of errgroup.